### PR TITLE
Removed coverage report from Travis Logs

### PR DESCRIPTION
**Pull request by git-abhishek**
## PR Summary
Initially I was showing `coverage report` at Travis interface due to testing purpose; now I think it is not needed to display there. We can exclusively check the coverage at our [Codecov](https://codecov.io/gh/yt-project/yt) page.

### CODE REVIEW

**Comment by brittonsmith** [comment]
I don't mind seeing the coverage report on PRs. I especially like seeing the change in coverage percentage. It's a good reminder to add tests.

**Comment by ngoldbaum** [comment]
@brittonsmith in case it wasn't clear, the codecov.io coverage reports will still show up in the "checks" area below the PR, they just won't show up in the Travis build logs. @git-abhishek please correct me if that's incorrect.

**Comment by git-abhishek** [comment]
Yes, @ngoldbaum that's correct. I am just removing from the Travis build logs, however it will still show up at the PR page.

**Comment by ngoldbaum** [comment]
I'm not sure why the flake8 tests are failing. flake8 should always check setup.cfg for ignore and exclude rules.

I wonder if it would be worthwhile to make the flake8 test a completely separate check and not part of the unit tests. While it would be nice to continue running flake8 as part of the test suite, the flakiness that we're seeing here argues that it would be better to run it outside the main unit test suite. For example, we could remove the flake8 test and set up stickler CI (https://stickler-ci.com/docs#python) or something like that so it's still enforced, but as a separate test that hopefully will reduce the flakiness of our unit tests.

**Comment by ngoldbaum** [comment]
I'd say +1 to approach 2. 

Did you take a look at stickler-ci.com? That's free for open source projects and we'd just be adding another checker separate from Travis.

**Comment by ngoldbaum** [comment]
@yt-fido test this please

**Comment by git-abhishek** [comment]
I did not do a deep dive on Stickler CI. I couldn't find quickly anyone elaborating the usage of Stickler. I thought that adding another external dependency just for flake8 tests would be too much of an overhead. However, let me know if Stickler CI offers more and if it is better to switch to it or its benefit outweighs the issues faced, I would be happy to check it out and integrate.

**Comment by ngoldbaum** [comment]
OK, let's stick with running the flak8 tests on travis as you have it set up right now. I'll hold off on merging this PR for a day or two to give @Xarthisius and @colinmarc time to weigh in.

**Comment by colinmarc** [comment]
Is there a reason flake8 has to complete before the tests start? Couldn't it run as a separate build stage/whatever, but parallel to the nosetests?

File: .travis.yml [hunk 1]
```diff
@@ -72,10 +54,49 @@ install:
...
+
+    - stage: tests
+      python: 2.7
+      env: NUMPY=numpy==1.10.4 CYTHON=cython==0.24 MATPLOTLIB=matplotlib==1.5.3 SYMPY=sympy==1.0 H5PY= SCIPY= FASTCACHE= FLAKE8= IPYTHON=ipython==1.0
+      script: coverage run $(which nosetests) -c nose.cfg yt
+
+    - stage: tests
+      python: 2.7
+      env: FLAKE8=
+      script: coverage run $(which nosetests) -c nose.cfg yt
```
Comment (Xarthisius) [inline .travis.yml]: what's the difference between this stage and the previous one?

Comment (ngoldbaum) [reply .travis.yml]: The previous one tests the minimum versions of our dependencies. This one tests python2.7 with the latest versions from pip.

**Comment by git-abhishek** [comment]
@colinmarc that's a nice suggestion. I guess this would approach would fall in the third category with [fast-finish](https://docs.travis-ci.com/user/customizing-the-build/#Fast-Finishing) turned on to be computationally efficient.

 Flake8 tests are completed first to enforce running of tests only when it is sanitized, thus saving us from wasteful computations of nosetests (as the build should fail finally). This would also serve as a minimum criterion to pass the build. I guess we would do this (flake8+nosetests, in parallel) if we are not concerned with doing these computations that might get rejected in future due to flake8 failure. 

We can also do it without `fast-finish` so that we know all the problems with the PR (lint+test failures) at once which could prevent from multiple commits.

**Comment by ngoldbaum** [comment]
I think failing fast with just the flake8 failure will be helpful. Often contributors don't lint their code and the way it's currently set up they don't get a message about the linting issues until after the tests fully complete ~45 minutes later. This way they'll see much more quickly what the issue is.

**Comment by git-abhishek** [comment]
@ngoldbaum in this PR, the current implemented strategy fails the build as soon as flake8 fails (but nosetests won't start until flake8 passes). Should I run flake8 in parallel to nosetests or sequential (flake8 then all the nosetests in parallel) as done already?

**Comment by ngoldbaum** [comment]
I'd personally say keep it as-is.

**Pull request merged**